File: ws/sanchez/u2/ws13ArraysJson/StudentsAppPy/model/json_manager.py
import json
import os
from model.student import Student

FILE_NAME = "student.json"

class JsonManager:
    @staticmethod
    def save_students(students):
        try:
            data = []
            for student in students:
                data.append({
                    "first_name": student.get_first_name(),
                    "last_name": student.get_last_name(),
                    "subject": student.get_subject(),
                    "age": student.get_age(),
                    "grades": student.get_grades()
                })
            
            with open(FILE_NAME, "w", encoding="utf-8") as file:
                json.dump(data, file, indent=4, ensure_ascii=False)
            print("\nData saved successfully!")
        except Exception as e:
            print(f"Error saving file: {e}")

    @staticmethod
    def read_students():
        if not os.path.exists(FILE_NAME):
            return []

        try:
            with open(FILE_NAME, "r", encoding="utf-8") as file:
                data = json.load(file)
                students = []
                for item in data:
                    student = Student(
                        item.get("first_name", ""),
                        item.get("last_name", ""),
                        item.get("subject", ""),
                        item.get("age", 0),
                        item.get("grades", [])
                    )
                    students.append(student)
                return students
        except Exception as e:
            print(f"No file found or error reading file: {e}")
            return []

File: ws/sanchez/u2/ws13ArraysJson/StudentsAppJS/model/jsonManager.js
const fs = require('fs');
const path = require('path');
const Student = require('./Student');

const FILE_NAME = path.join(__dirname, '..', 'student.json');

function saveStudents(students) {
    try {
        const data = [];
        for (let s of students) {
            data.push({
                first_name: s.firstName,
                last_name: s.lastName,
                subject: s.subject,
                age: s.age,
                grades: s.grades
            });
        }
        fs.writeFileSync(FILE_NAME, JSON.stringify(data, null, 4));
        console.log("\nData saved successfully!");
    } catch (e) {
        console.log(`Error saving file: ${e.message}`);
    }
}

function readStudents() {
    if (!fs.existsSync(FILE_NAME)) return [];
    try {
        const data = JSON.parse(fs.readFileSync(FILE_NAME, 'utf8'));
        const students = [];
        for (let item of data) {
            students.push(new Student(
                item.first_name,
                item.last_name,
                item.subject,
                item.age,
                item.grades
            ));
        }
        return students;
    } catch (e) {
        console.log(`Error reading file: ${e.message}`);
        return [];
    }
}

module.exports = { saveStudents, readStudents };